samples/local_attestation/common/attestation.cpp

```diff
@@ -2,164 +2,211 @@
  * following three steps:
  *
- * 1) The local report is first attested using the oe_verify_report API. This
- * ensures the authenticity of the enclave that generated the report.
- * 2) Next, to establish trust of the enclave that generated the report,
- * the mrsigner, product_id, isvsvn values are checked to  see if they are
+ * 1) The local evidence is first attested using the oe_verify_evidence API.
+ * This ensures the authenticity of the enclave that generated the evidence. 2)
+ * Next, to establish trust of the enclave that generated the evidence, the
+ * mrsigner, product_id, isvsvn values are checked to  see if they are
  * predefined trusted values.
- * 3) Once the enclave's trust has been established, the validity of
- * accompanying data is ensured by comparing its SHA256 digest against the
- * report_data field.
  */
-bool Attestation::attest_local_report(
-    const uint8_t* local_report,
-    size_t report_size,
+bool Attestation::attest_local_evidence(
+    const uint8_t* local_evidence,
+    size_t evidence_size,
     const uint8_t* data,
     size_t data_size)
 {
     bool ret = false;
     uint8_t sha256[32];
-    oe_report_t parsed_report = {0};
     oe_result_t result = OE_OK;
+    oe_result_t verifier_result = OE_OK;
+    oe_claim_t* claims = NULL;
+    size_t claims_length = 0;
 
-    // While attesting, the report being attested must not be tampered
+    // While attesting, the evidence being attested must not be tampered
     // with. Ensure that it has been copied over to the enclave.
-    if (!oe_is_within_enclave(local_report, report_size))
+    if (!oe_is_within_enclave(local_evidence, evidence_size))
     {
-        TRACE_ENCLAVE("Cannot attest report in host memory. Unsafe.");
+        TRACE_ENCLAVE("Cannot attest evidence in host memory. Unsafe.");
         goto exit;
     }
 
-    TRACE_ENCLAVE("report_size = %ld", report_size);
+    TRACE_ENCLAVE("evidence_size = %ld", evidence_size);
 
-    // 1)  Validate the report's trustworthiness
-    // Verify the report to ensure its authenticity.
-    result = oe_verify_report(local_report, report_size, &parsed_report);
-    if (result != OE_OK)
+    verifier_result = oe_verifier_initialize();
+    if (verifier_result != OE_OK)
     {
-        TRACE_ENCLAVE("oe_verify_report failed (%s).\n", oe_result_str(result));
+        TRACE_ENCLAVE("oe_verifier_initialize failed.");
         goto exit;
     }
 
-    TRACE_ENCLAVE("oe_verify_report succeeded\n");
-
-    // 2) validate the enclave identity's signed_id is the hash of the public
-    // signing key that was used to sign an enclave. Check that the enclave was
-    // signed by an trusted entity.
-    if (memcmp(parsed_report.identity.signer_id, m_enclave_mrsigner, 32) != 0)
+    // 1)  Validate the evidence's trustworthiness
+    // Verify the evidence to ensure its authenticity.
+    result = oe_verify_evidence(
+        &sgx_local_uuid,
+        local_evidence,
+        evidence_size,
+        NULL,
```

Comment:
suggestion: this is a CPP file so use the C++ keyword "nullptr" instead of "NULL"